tests/ci_build/Dockerfile.gpu_dev_ver

```
@@ -0,0 +1,51 @@
+# Container to test XGBoost against dev versions of dependencies
+
+ARG CUDA_VERSION_ARG
+FROM nvidia/cuda:$CUDA_VERSION_ARG-runtime-ubuntu22.04
+ARG CUDA_VERSION_ARG
+ARG RAPIDS_VERSION_ARG
+ARG NCCL_VERSION_ARG
+
+# Environment
+ENV DEBIAN_FRONTEND noninteractive
+SHELL ["/bin/bash", "-c"]   # Use Bash as shell
+
+# Install all basic requirements
+RUN \
+    apt-key adv --fetch-keys https://developer.download.nvidia.com/compute/cuda/repos/ubuntu2204/x86_64/3bf863cc.pub && \
+    apt-get update && \
+    apt-get install -y wget unzip bzip2 libgomp1 build-essential openjdk-8-jdk-headless && \
+    # Python
+    wget -nv -O conda.sh https://github.com/conda-forge/miniforge/releases/download/22.11.1-2/Mambaforge-22.11.1-2-Linux-x86_64.sh && \
+    bash conda.sh -b -p /opt/mambaforge
+
+ENV PATH=/opt/mambaforge/bin:$PATH
+
+# Create new Conda environment with cuDF, Dask, and cuPy
+RUN \
+    export NCCL_SHORT_VER=$(echo "$NCCL_VERSION_ARG" | cut -d "-" -f 1) && \
+    mamba create -y -n gpu_test -c rapidsai-nightly -c nvidia -c conda-forge \
+        python=3.10 "cudf>$RAPIDS_VERSION_ARG" "rmm>$RAPIDS_VERSION_ARG" cudatoolkit=$CUDA_VERSION_ARG \
```

RAPIDS needs a different channel order. Also, why are we using `>` here instead of `=`?
```suggestion
    mamba create -y -n gpu_test -c rapidsai-nightly -c conda-forge -c nvidia \
        python=3.10 "cudf=$RAPIDS_VERSION_ARG" "rmm=$RAPIDS_VERSION_ARG" cudatoolkit=$CUDA_VERSION_ARG \
```

---

I'm using `>` here because the variable `$RAPIDS_VERSION_ARG` indicates the latest stable version. https://github.com/dmlc/xgboost/blob/d2d01d977a5ba07e55ed9230e2bd83539e81d184/tests/buildkite/conftest.sh#L27

---

Let me add a comment here.

---

You probably want a solution that will use the _nightly_ version like 24.06 rather than the _previous stable_ version like 24.04.

If you do that, you can use `=X.Y.*`. That's how we handle it in RAPIDS:

```
- librmm==24.8.*
```

https://github.com/rapidsai/cudf/blob/1354abdb7a4f9eb58bfc6e359c49d0baabacb4e1/conda/environments/all_cuda-122_arch-x86_64.yaml#L49

---

@bdice Didn't realize that there might be two nightly versions. I updated the PR to specify the nightly version exactly.